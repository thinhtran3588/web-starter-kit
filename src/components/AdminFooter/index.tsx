import React from 'react';
import { withTranslation, WithTranslation } from '@app/core';
import { config } from '@app/config';
import { Typography } from '../Typography';
import { Link } from '../Link';
import { useStyles } from './styles';

type Props = WithTranslation;

export const BaseAdminFooter = (props: Props): JSX.Element => {
  const { t } = props;
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      <Typography variant='body1'>
        &copy; <Link href='/'>{config.author}</Link>
        &nbsp;{config.copyRightYear}
      </Typography>
      <Typography variant='caption'>{t('createdWithLove')}</Typography>
    </footer>
  );
};

export const AdminFooter = withTranslation('admin_common')(BaseAdminFooter);
