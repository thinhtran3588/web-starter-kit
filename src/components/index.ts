export * from './AdminLayout';
export * from './AdminFooter';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Layout';
export * from './Sidebar';
export * from './SidebarMenu';
export * from './FormHeader';
export * from './Field';
export * from './FormFilter';
export * from './FormInput';
export * from './FormSearch';
export * from './Table';
export * from './LanguageSelection';
export * from './Link';
export * from './AuthLayout';
export * from './Head';
export * from './Button';
export * from './Grid';
export * from './Typography';
