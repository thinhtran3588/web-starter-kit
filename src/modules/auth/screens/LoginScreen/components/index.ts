export * from './EmailLogin';
export * from './PhoneNoLogin';
