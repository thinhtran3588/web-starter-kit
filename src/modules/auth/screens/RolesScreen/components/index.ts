export * from './Detail';
export * from './PermissionsTable';
