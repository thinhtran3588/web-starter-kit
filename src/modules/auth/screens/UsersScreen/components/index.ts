export * from './Detail';
export * from './PermissionsTable';
export * from './ChangePassword';
