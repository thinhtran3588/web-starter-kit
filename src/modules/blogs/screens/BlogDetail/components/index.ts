export * from './Content';
export * from './SideMenu';
