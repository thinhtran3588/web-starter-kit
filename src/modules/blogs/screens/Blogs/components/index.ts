export * from './Categories';
export * from './BlogsList';
