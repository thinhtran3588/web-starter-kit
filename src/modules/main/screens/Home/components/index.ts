export * from './CoverImage';
export * from './Features';
export * from './Quote';
export * from './Help';
