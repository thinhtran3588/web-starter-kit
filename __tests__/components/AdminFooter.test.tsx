import React from 'react';
import { render } from '@testing-library/react';
import { AdminFooter } from '@app/components';

describe('components/AdminFooter', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(<AdminFooter />);

    expect(baseElement).toMatchSnapshot();
  });
});
