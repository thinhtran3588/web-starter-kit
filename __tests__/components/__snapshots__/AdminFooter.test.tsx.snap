// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AdminFooter renders successfully 1`] = `
<body>
  <div>
    <footer
      class="makeStyles-root-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        © 
        <next-link
          href="/"
        >
          <a
            class=""
            color="inherit"
            href="/"
          >
            Thinh Tran
          </a>
        </next-link>
         
        2019
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption"
      >
        admin_common.createdWithLove
      </span>
    </footer>
  </div>
</body>
`;
